README.md

```diff
@@ -1,34 +1,145 @@
+
+### 1.2 Create and manage Jira issues in Mattermost
+
+Connect your Mattermost account to Jira via `/jira connect` command, then create and manage issues across Mattermost channels. You can disconnect your account anytime via `/jira disconnect`.
+
+#### 1.2.1 Create Jira issues
+
+Create Jira issues from a Mattermost message by clicking the **More Actions** (...) option of any message in the channel (available when you hover over a message), then selecting **Create Jira Issue**.
+
+Then, on the resulting issue creation dialog, select the project, issue type and enter other fields to create the issue.
+
+![image](https://user-images.githubusercontent.com/13119842/59113188-985a9280-8912-11e9-9def-9a7382b4137e.png)
+
+Click **Create** and the Jira issue is now created, including any file attachments part of the Mattermost message.
+
+![image](https://user-images.githubusercontent.com/13119842/59113219-a4deeb00-8912-11e9-9741-5ddc8a4b51fa.png)
+
+**NOTE**: This plugin does not support all Jira fields. If the project you tried to create an issue for has **required fields** not yet supported, you will be prompted to manually create an issue. Clicking the provided link brings you to an issue creation screen in Jira, with the fields entered previously pre-filled.
+
+The supported Jira fields are:
+
+  - **Project Picker**: Custom fields and the built-in **Project** field.
+  - **Single-Line Text**: Custom fields, and built-in fields such as **Summary**, **Environment**.
+  - **Multi-Line Text**: Custom fields, and built-in fields such as **Description**.
+  - **Single-Choice Issue**: Custom fields, and built-in fields such as **Issue Type** and **Priority**. 
 
-The above are supported for Jira Core and Jira Software, both for Cloud and Server platforms.
+#### 1.2.2 Attach Messages to Jira Issues
 
-## 2. Jira Plugin v2 Roadmap
+Keep all information in one place by attaching parts of Mattermost conversations in Jira issues as comments. To attach a message, click the **More Actions** (...) option of any message in the channel (available when you hover over a message), then select **Attach to Jira Issue**.
+
+Then, on the resulting dialog, select issue you want to attach it to. You may search for issues containing specific text.
+
+![image](https://user-images.githubusercontent.com/13119842/59113267-b627f780-8912-11e9-90ec-417d430de7e6.png)
+
+Click **Attach** and the message is attached to the selected Jira issue as a comment.
+
+#### 1.2.3 Transition Jira issues
+
+Transition issues without the need to switch to your Jira project. To transition an issue, use the `/jira transition <issue-key> <state>` command.
+
+For instance, `/jira transition EXT-20 done` transitions the issue key **EXT-20** to **Done**.
+
+![image](https://user-images.githubusercontent.com/13119842/59113377-dfe11e80-8912-11e9-8971-f869fa123366.png)
+
+Note that states and issue transitions are based on your Jira project workflow configuration.
```

Comment:
What happens when a user types in a state that is not valid (ie: I use "submitted" but "submitted" does not exist in the workflow configured for that particular project?

Reply:
An ephemeral message is posted back with text `We couldn't find the state. Please use a Jira state such as 'done' and try again.`

Added a note about this in the doc